refactor(language_server): new configuration structure for `initialize` and `workspace/didChangeConfiguration`

Sysix opened the pull request:
This PR introduces a new Options structure, which will be used for `initialize` and `workspace/didChangeConfiguration`.
This is needed for multi workspace support, or else we will do the less performance-way with sending a server-request to the client.

Updated the docs to reflect how to use the new configuration structure
Still used the old structure as a fallback, because the IntelliJ plugin ❤️

Sysix commented:
* **#10890** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10890?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10890?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10890).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented:
@nrayburn-tech I tested it with VSCode extension in main and the language server in this PR.
Everything should still work for the IntelliJ Plugin. 
Could you guys start using the new format so we can remove the fallbacks in the future ❤️ ?

nrayburn-tech commented:
> @nrayburn-tech I tested it with VSCode extension in main and the language server in this PR. Everything should still work for the IntelliJ Plugin. Could you guys start using the new format so we can remove the fallbacks in the future ❤️ ?

I haven’t tested or reviewed anything, but I don’t think migrating should be difficult if I just need to change the input arguments to the initialize command.

Sysix commented:
I wanted to wait for all PRs to be approved. 
This change should be in main / next release as fast as possible → merging

Sysix committed:
refactor(language_server): new configuration structure for `initialize` and `workspace/didChangeConfiguration` (#10890)

This PR introduces a new Options structure, which will be used for `initialize` and `workspace/didChangeConfiguration`.
This is needed for multi workspace support, or else we will do the less performance-way with sending a server-request to the client.

Updated the docs to reflect how to use the new configuration structure
Still used the old structure as a fallback, because the IntelliJ plugin ❤️
